### PR TITLE
174610297 first ten questions

### PR DESCRIPTION
**Pull request by NickCallaghan**
**What type of PR is this? (check all applicable)**
☐ :recycle: Refactor
☑ :sparkles: Feature
☐ :bug: Bug Fix
☐ :construction_worker: Optimization
☐ :memo: Documentation Update
☐ :bookmark: Release
☐ :triangular_flag_on_post: Other

**Description**
First 10 Questions of the Questionnaire
- Added Quiz component to be Questionnaire and changed url to /questionnaire
- added load in the questions and remove a question from the ones to be answered once it have been answered
- added a Question Component
- added a loading spinner to be shown if the questions are not yet loaded

**Not in scope**
- Not currently doing anything with the users answers will need to deal with that later
- There are a few styling tweaks thats are needed to match the design but we can come back to those
- backend has been requested to return the answer as an array of objects {id, answer}. We'll need too update once this happens. 

**Related Tickets & Documents**
#174610297 - [First 10 Questions](https://www.pivotaltracker.com/story/show/174610297)

**Added tests?**
☐ :+1: yes
☐ :no_good: no, because they aren't needed
☑ :raising_hand: no, because I need help
Added tests for the question components and also added a few snapshots. Almost having second thoughts about the snapshots my remove? Thoughts?

**Added to documentation?**
☐ :scroll: readme
☑ :no_good: no documentation needed
Screenshot 2020-09-06 at 00 55 18
Screenshot 2020-09-06 at 00 58 47

### CODE REVIEW

File: src/pages/Questionnaire.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  );
+  // Current Question being answered
+  const [currentQuestion, setCurrentQuestion] = useState<TQuestion | null>(
+    null
+  );
+  // Question number user is on
+  const [progress, setProgress] = useState(1);
+
+  // Pick a random question from the Questions to answer
+  const changeQuestion = useCallback(() => {
```
Comment (thiyagarajan-angappan) [inline src/pages/Questionnaire.tsx:34]: @NickCallaghan should we try using a stack structure so could do a pop() and puss() while implementing a Back button?

Comment (NickCallaghan) [reply src/pages/Questionnaire.tsx:34]: Hi @thiyagarajan-angappan I did think about the back functionality (after i wrote this). I reckon you are right but my plan was just to re-write it when the time comes to deal with the back button. 

We need to randomise the questions each time so thinking maybe deal with that within the useQuestions Hook? Let me know what you think?

Comment (johols) [reply src/pages/Questionnaire.tsx:34]: The randomization can be moved to useQuestions, that might give cleaner code in the components also

Comment (NickCallaghan) [reply src/pages/Questionnaire.tsx:34]: Can we say then that we'll move that within the a subsequent story we re-write the internals of this component to use a stack for the questions and move the randomisation of the questions into the hook?

Comment (thiyagarajan-angappan) [reply src/pages/Questionnaire.tsx:34]: Sorry could not get to github yesterday... was not at home ! We can do it with another story. Just wanted to mark it for change

**Review by thiyagarajan-angappan — APPROVED** [review]

File: src/pages/Questionnaire.tsx (hunk 1 above)
Comment (NickCallaghan) [reply src/pages/Questionnaire.tsx:34]: no worries all good. :) 

TBH I didn't give the back functionality any consideration until after i wrote the code.

**Pull request merged**